src/lib/y2network/interface.rb

```diff
@@ -95,6 +95,11 @@ def ==(other)
     # https://ruby-doc.org/core-2.3.3/Object.html#method-i-eql-3F
     alias_method :eql?, :==
 
+    # Used by Array or Hash in order to compare equality of elements (bsc#1186082)
+    def hash
+      name.hash
```

Comment:
~~~
name = "foo"
interface = Interface.new(name)

interface.hash == name.hash #=> true
~~~

I would use the hash of the class too to make sure we don't have false positives with another kind of objects.

Reply:
In general I agree, in this case the **==** method do not contain the class check neither. It would be hard to combine that kinds of objects but it is true that is open to that kind of errors.

Reply:
Yep, although it is strange to combine that kind of objects, it is open to that errors and the **==** already contains that check, so will take that into account in next iteration when implemented generally.